Classes/Utility/TranslationLabelUtility.php

```diff
@@ -138,34 +141,73 @@ public static function renderTranslationWithExtendedInformation(
     /**
      * @param string $labelKey
      * @param string $extensionName
+     * @param RenderingContextInterface $renderingContext
      * @return bool
      * @throws AspectNotFoundException
      */
-    public static function isFrontendWithLoggedInBEUser($labelKey = '', $extensionName = '')
+    public static function meetsRenderingConditionsForExtendedInformation($labelKey = '', $extensionName = '', $renderingContext = null)
     {
-        $showTranslationLabels = false;
+        if (!$labelKey) {
+            $labelKey = '';
+        }
+        if (!$extensionName) {
+            $extensionName = '';
+        }
+
+        $showTranslationLabels = self::checkRenderingConditionsForExtendedInformation($labelKey, $extensionName);
+
+        $eventDispatcher = GeneralUtility::makeInstance(EventDispatcherInterface::class);
+        /** @var CheckedRenderingConditionsEvent $event */
+        $event = $eventDispatcher->dispatch(
+            new CheckedRenderingConditionsEvent($showTranslationLabels, $labelKey, $extensionName, $renderingContext)
+        );
+
+        $showTranslationLabels = $event->getShowTranslationLabels();
```

Comment:
Even listeners have the ability to enable translation labels even a previous condition failed. This gives full control over the event, but could be dangerous.

Do we want this or would it be better not to send the event if a basic check failed?